packages/go_router/test/go_route_information_parser_test.dart

```diff
@@ -0,0 +1,192 @@
+// Copyright 2013 The Flutter Authors. All rights reserved.
+// Use of this source code is governed by a BSD-style license that can be
+// found in the LICENSE file.
+
+import 'package:flutter/material.dart';
+import 'package:flutter_test/flutter_test.dart';
+import 'package:go_router/go_router.dart';
+import 'package:go_router/src/go_route_information_parser.dart';
+import 'package:go_router/src/go_route_match.dart';
+
+void main() {
+  test('GoRouteInformationParser can parse route', () async {
```

Comment:
Should there be tests on malformed inputs? One weird example would be something like `RouteInformation(location: '/hello?foo=bar&foo=buzz')

Reply:
Yeah I can add a test for that.